src/main/java/com/uber/cadence/internal/common/WorkflowExecutionUtils.java

```diff
@@ -311,7 +311,19 @@ private static HistoryEvent getInstanceCloseEvent(
                         + ", unit="
                         + unit));
           }
-          History history = r.getHistory();
```

Comment:
Great catch Bowei, thanks!
I believe the other response.getHistory usages needs to be updated the same way: e.g. https://github.com/uber/cadence-java-client/blob/5b28096165ee4bb075bb1a9ac10064916b7b070a/src/main/java/com/uber/cadence/internal/worker/WorkflowWorker.java#L131
https://github.com/uber/cadence-java-client/blob/243e758a0093a5916ec08f73a02c580ac2af52b1/src/main/java/com/uber/cadence/internal/common/WorkflowExecutionUtils.java#L248
.....

Reply:
Thanks for bring it up! 
I double checked and found they actually don't need to be updated because https://github.com/uber/cadence-java-client/pull/472 already handle it for sync `getWorkflowExecutionHistory`. 
but it help me realize that `WorkflowExecutionUtils` is not a good place as `WorkflowServiceTChannel`, so I move my fix there.